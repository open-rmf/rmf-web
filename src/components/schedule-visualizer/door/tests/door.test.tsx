import * as RomiCore from '@osrf/romi-js-core-interfaces';
import { createMount } from '@material-ui/core/test-utils';
import { ReactWrapper } from 'enzyme';
import React from 'react';
import SingleHingeDoor from '../door-single-hinge';
import SingleSlideDoor from '../door-single-slide';
import DefaultDoor from '../door-default';

const mainDoor = {
  name: 'main_door',
  v1_x: 8.2,
  v1_y: -5.5,
  v2_x: 7.85,
  v2_y: -6.2,
  door_type: 2,
  motion_range: -1.571,
  motion_direction: 1,
};

const mount = createMount();

test('Trigger click event', async () => {
  let clicked = false;
  const handleClick = () => {
    clicked = true;
  };
  const wrapper = mount(
    <svg>
      <DefaultDoor
        v1={[mainDoor.v1_x, mainDoor.v1_y]}
        v2={[mainDoor.v2_x, mainDoor.v2_y]}
        onClick={handleClick}
      />
    </svg>,
  );
  wrapper
    .find(DefaultDoor)
    .at(0)
    .find('g')
    .simulate('click');
  expect(clicked).toBe(true);

  wrapper.unmount();
});

describe('Checks assignation of styles on different states of SingleHingeDoor', () => {
  const getSingleDoorSVGLine = (wrapper: ReactWrapper) => {
    return wrapper
      .find(DefaultDoor)
      .at(0)
      .find('line')
      .at(0);
  };

  test('Style of SingleHingeDoor on Open', () => {
    const wrapper = mount(
      <svg>
        <SingleHingeDoor
          v1={[mainDoor.v1_x, mainDoor.v1_y]}
          v2={[mainDoor.v2_x, mainDoor.v2_y]}
          door={mainDoor}
          currentMode={RomiCore.DoorMode.MODE_OPEN}
        />
      </svg>,
    );
    const singleDoorSVGLine = getSingleDoorSVGLine(wrapper);
    expect(singleDoorSVGLine.hasClass(/(makeStyles)-(doorOpen)-(\d+)/)).toBe(true);
    wrapper.unmount();
  });

  test('Style of SingleHingeDoor on Close', () => {
    const wrapper = mount(
      <svg>
        <SingleHingeDoor
          v1={[mainDoor.v1_x, mainDoor.v1_y]}
          v2={[mainDoor.v2_x, mainDoor.v2_y]}
          door={mainDoor}
          currentMode={RomiCore.DoorMode.MODE_CLOSED}
        />
      </svg>,
    );
    const singleDoorSVGLine = getSingleDoorSVGLine(wrapper);
    expect(singleDoorSVGLine.hasClass(/(makeStyles)-(doorClose)-(\d+)/)).toBe(true);
    wrapper.unmount();
  });

  test('Style of SingleHingeDoor on Process', () => {
    const wrapper = mount(
      <svg>
        <SingleSlideDoor
          v1={[mainDoor.v1_x, mainDoor.v1_y]}
          v2={[mainDoor.v2_x, mainDoor.v2_y]}
          door={mainDoor}
          currentMode={RomiCore.DoorMode.MODE_MOVING}
        />
      </svg>,
    );
    const singleDoorSVGLine = getSingleDoorSVGLine(wrapper);
    expect(singleDoorSVGLine.hasClass(/(makeStyles)-(doorProcess)-(\d+)/)).toBe(true);
    wrapper.unmount();
  });
});
