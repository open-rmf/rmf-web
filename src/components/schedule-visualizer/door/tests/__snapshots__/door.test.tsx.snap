// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Doors renders correctly 1`] = `
<svg>
  <SingleSlideDoor
    currentMode={2}
    door={
      Object {
        "door_type": 2,
        "motion_direction": 1,
        "motion_range": -1.571,
        "name": "main_door",
        "v1_x": 8.2,
        "v1_y": -5.5,
        "v2_x": 7.85,
        "v2_y": -6.2,
      }
    }
    v1={
      Array [
        8.2,
        -5.5,
      ]
    }
    v2={
      Array [
        7.85,
        -6.2,
      ]
    }
  >
    <DefaultDoor
      style="makeStyles-doorOpen-11"
      v1={
        Array [
          8.2,
          5.5,
        ]
      }
      v2={
        Array [
          7.85,
          6.2,
        ]
      }
    >
      <g
        onClick={[Function]}
      >
        <line
          className="makeStyles-doorMarker-8 makeStyles-door-9 makeStyles-doorOpen-11"
          x1={8.2}
          x2={7.85}
          y1={5.5}
          y2={6.2}
        />
      </g>
    </DefaultDoor>
    <DefaultDoor
      style="makeStyles-doorTransparent-14"
      v1={
        Array [
          8.2,
          5.5,
        ]
      }
      v2={
        Array [
          7.85,
          6.2,
        ]
      }
    >
      <g
        onClick={[Function]}
      >
        <line
          className="makeStyles-doorMarker-8 makeStyles-door-9 makeStyles-doorTransparent-14"
          x1={8.2}
          x2={7.85}
          y1={5.5}
          y2={6.2}
        />
      </g>
    </DefaultDoor>
  </SingleSlideDoor>
</svg>
`;

exports[`Doors renders correctly 2`] = `
<svg>
  <DoubleSlideDoor
    currentMode={2}
    door={
      Object {
        "door_type": 2,
        "motion_direction": 1,
        "motion_range": -1.571,
        "name": "main_door",
        "v1_x": 8.2,
        "v1_y": -5.5,
        "v2_x": 7.85,
        "v2_y": -6.2,
      }
    }
    v1={
      Array [
        8.2,
        -5.5,
      ]
    }
    v2={
      Array [
        7.85,
        -6.2,
      ]
    }
  >
    <SingleSlideDoor
      currentMode={2}
      door={
        Object {
          "door_type": 2,
          "motion_direction": 1,
          "motion_range": -1.571,
          "name": "main_door",
          "v1_x": 8.2,
          "v1_y": -5.5,
          "v2_x": 7.85,
          "v2_y": -6.2,
        }
      }
      v1={
        Array [
          8.2,
          -5.5,
        ]
      }
      v2={
        Array [
          8.024999999999999,
          -5.85,
        ]
      }
    >
      <DefaultDoor
        style="makeStyles-doorOpen-18"
        v1={
          Array [
            8.2,
            5.5,
          ]
        }
        v2={
          Array [
            8.024999999999999,
            5.85,
          ]
        }
      >
        <g
          onClick={[Function]}
        >
          <line
            className="makeStyles-doorMarker-15 makeStyles-door-16 makeStyles-doorOpen-18"
            x1={8.2}
            x2={8.024999999999999}
            y1={5.5}
            y2={5.85}
          />
        </g>
      </DefaultDoor>
      <DefaultDoor
        style="makeStyles-doorTransparent-21"
        v1={
          Array [
            8.2,
            5.5,
          ]
        }
        v2={
          Array [
            8.024999999999999,
            5.85,
          ]
        }
      >
        <g
          onClick={[Function]}
        >
          <line
            className="makeStyles-doorMarker-15 makeStyles-door-16 makeStyles-doorTransparent-21"
            x1={8.2}
            x2={8.024999999999999}
            y1={5.5}
            y2={5.85}
          />
        </g>
      </DefaultDoor>
    </SingleSlideDoor>
    <SingleSlideDoor
      currentMode={2}
      door={
        Object {
          "door_type": 2,
          "motion_direction": 1,
          "motion_range": -1.571,
          "name": "main_door",
          "v1_x": 8.2,
          "v1_y": -5.5,
          "v2_x": 7.85,
          "v2_y": -6.2,
        }
      }
      v1={
        Array [
          8.024999999999999,
          -5.85,
        ]
      }
      v2={
        Array [
          7.85,
          -6.2,
        ]
      }
    >
      <DefaultDoor
        style="makeStyles-doorOpen-18"
        v1={
          Array [
            8.024999999999999,
            5.85,
          ]
        }
        v2={
          Array [
            7.85,
            6.2,
          ]
        }
      >
        <g
          onClick={[Function]}
        >
          <line
            className="makeStyles-doorMarker-15 makeStyles-door-16 makeStyles-doorOpen-18"
            x1={8.024999999999999}
            x2={7.85}
            y1={5.85}
            y2={6.2}
          />
        </g>
      </DefaultDoor>
      <DefaultDoor
        style="makeStyles-doorTransparent-21"
        v1={
          Array [
            8.024999999999999,
            5.85,
          ]
        }
        v2={
          Array [
            7.85,
            6.2,
          ]
        }
      >
        <g
          onClick={[Function]}
        >
          <line
            className="makeStyles-doorMarker-15 makeStyles-door-16 makeStyles-doorTransparent-21"
            x1={8.024999999999999}
            x2={7.85}
            y1={5.85}
            y2={6.2}
          />
        </g>
      </DefaultDoor>
    </SingleSlideDoor>
  </DoubleSlideDoor>
</svg>
`;

exports[`Doors renders correctly 3`] = `
<svg>
  <SingleHingeDoor
    currentMode={2}
    door={
      Object {
        "door_type": 2,
        "motion_direction": 1,
        "motion_range": -1.571,
        "name": "main_door",
        "v1_x": 8.2,
        "v1_y": -5.5,
        "v2_x": 7.85,
        "v2_y": -6.2,
      }
    }
    v1={
      Array [
        8.2,
        -5.5,
      ]
    }
    v2={
      Array [
        7.85,
        -6.2,
      ]
    }
  >
    <DefaultDoor
      style="makeStyles-doorOpen-25"
      v1={
        Array [
          8.2,
          5.5,
        ]
      }
      v2={
        Array [
          7.85,
          6.2,
        ]
      }
    >
      <g
        onClick={[Function]}
      >
        <line
          className="makeStyles-doorMarker-22 makeStyles-door-23 makeStyles-doorOpen-25"
          x1={8.2}
          x2={7.85}
          y1={5.5}
          y2={6.2}
        />
      </g>
    </DefaultDoor>
    <DefaultDoor
      style="makeStyles-doorTransparent-28"
      v1={
        Array [
          8.2,
          5.5,
        ]
      }
      v2={
        Array [
          7.85,
          6.2,
        ]
      }
    >
      <g
        onClick={[Function]}
      >
        <line
          className="makeStyles-doorMarker-22 makeStyles-door-23 makeStyles-doorTransparent-28"
          x1={8.2}
          x2={7.85}
          y1={5.5}
          y2={6.2}
        />
      </g>
    </DefaultDoor>
  </SingleHingeDoor>
</svg>
`;

exports[`Doors renders correctly 4`] = `
<svg>
  <DoubleHingeDoor
    currentMode={2}
    door={
      Object {
        "door_type": 2,
        "motion_direction": 1,
        "motion_range": -1.571,
        "name": "main_door",
        "v1_x": 8.2,
        "v1_y": -5.5,
        "v2_x": 7.85,
        "v2_y": -6.2,
      }
    }
    v1={
      Array [
        8.2,
        -5.5,
      ]
    }
    v2={
      Array [
        7.85,
        -6.2,
      ]
    }
  >
    <SingleHingeDoor
      currentMode={2}
      door={
        Object {
          "door_type": 2,
          "motion_direction": 1,
          "motion_range": -1.571,
          "name": "main_door",
          "v1_x": 8.2,
          "v1_y": -5.5,
          "v2_x": 7.85,
          "v2_y": -6.2,
        }
      }
      v1={
        Array [
          8.2,
          -5.5,
        ]
      }
      v2={
        Array [
          8.024999999999999,
          -5.85,
        ]
      }
    >
      <DefaultDoor
        style="makeStyles-doorOpen-32"
        v1={
          Array [
            8.2,
            5.5,
          ]
        }
        v2={
          Array [
            8.024999999999999,
            5.85,
          ]
        }
      >
        <g
          onClick={[Function]}
        >
          <line
            className="makeStyles-doorMarker-29 makeStyles-door-30 makeStyles-doorOpen-32"
            x1={8.2}
            x2={8.024999999999999}
            y1={5.5}
            y2={5.85}
          />
        </g>
      </DefaultDoor>
      <DefaultDoor
        style="makeStyles-doorTransparent-35"
        v1={
          Array [
            8.2,
            5.5,
          ]
        }
        v2={
          Array [
            8.024999999999999,
            5.85,
          ]
        }
      >
        <g
          onClick={[Function]}
        >
          <line
            className="makeStyles-doorMarker-29 makeStyles-door-30 makeStyles-doorTransparent-35"
            x1={8.2}
            x2={8.024999999999999}
            y1={5.5}
            y2={5.85}
          />
        </g>
      </DefaultDoor>
    </SingleHingeDoor>
    <SingleHingeDoor
      currentMode={2}
      door={
        Object {
          "door_type": 2,
          "motion_direction": 1,
          "motion_range": -1.571,
          "name": "main_door",
          "v1_x": 8.2,
          "v1_y": -5.5,
          "v2_x": 7.85,
          "v2_y": -6.2,
        }
      }
      v1={
        Array [
          8.024999999999999,
          -5.85,
        ]
      }
      v2={
        Array [
          7.85,
          -6.2,
        ]
      }
    >
      <DefaultDoor
        style="makeStyles-doorOpen-32"
        v1={
          Array [
            8.024999999999999,
            5.85,
          ]
        }
        v2={
          Array [
            7.85,
            6.2,
          ]
        }
      >
        <g
          onClick={[Function]}
        >
          <line
            className="makeStyles-doorMarker-29 makeStyles-door-30 makeStyles-doorOpen-32"
            x1={8.024999999999999}
            x2={7.85}
            y1={5.85}
            y2={6.2}
          />
        </g>
      </DefaultDoor>
      <DefaultDoor
        style="makeStyles-doorTransparent-35"
        v1={
          Array [
            8.024999999999999,
            5.85,
          ]
        }
        v2={
          Array [
            7.85,
            6.2,
          ]
        }
      >
        <g
          onClick={[Function]}
        >
          <line
            className="makeStyles-doorMarker-29 makeStyles-door-30 makeStyles-doorTransparent-35"
            x1={8.024999999999999}
            x2={7.85}
            y1={5.85}
            y2={6.2}
          />
        </g>
      </DefaultDoor>
    </SingleHingeDoor>
  </DoubleHingeDoor>
</svg>
`;
